### PR TITLE
feat(ot3, api): Add gripper class 

### PR DESCRIPTION
**Pull request by ahiuchingau**
# Overview
💥💥💥 New instrument alert: the Gripper has arrived, well, not fully, but we're getting there. 💥💥💥

Note: We previously interchange the words "pipette" and "instrument" for the OT1 & OT2. This should no longer be the case for the OT3. Gripper is a type of instrument that we'll be attaching to the OT3; just like we would a pipette.

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

**Comment by Laura-Danielle** [comment]
Alt Text

File: api/src/opentrons/hardware_control/gripper.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        # cache a dict representation of config for improved performance of
+        # as_dict.
+        self._config_as_dict = asdict(config)
+
+    @property
+    def config(self) -> gripper_config.GripperConfig:
+        return self._config
+
+    def update_config_item(self, elem_name: str, elem_val: Any) -> None:
+        self._log.info("updated config: {}={}".format(elem_name, elem_val))
```
Comment (amitlissack) [inline api/src/opentrons/hardware_control/gripper.py]: nit: f-strings vs format

File: api/src/opentrons/hardware_control/gripper.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    def __str__(self) -> str:
+        return "{}".format(self._config.display_name)
+
+    def __repr__(self) -> str:
+        return "<{}: {} {}>".format(
+            self.__class__.__name__, self._config.display_name, id(self)
+        )
+
+    def as_dict(self) -> "Gripper.DictType":
```
Comment (amitlissack) [inline api/src/opentrons/hardware_control/gripper.py:83]: Why do we need an `as_dict` method?

File: api/src/opentrons/hardware_control/gripper.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    DictType = Dict[str, Union[str, float, bool]]
+    #: The type of this data class as a dict
+
+    def __init__(
+        self,
+        config: gripper_config.GripperConfig,
+        gripper_id: Optional[str] = None,
+    ) -> None:
+        self._config = config
+        self._name = self._config.name
```
Comment (amitlissack) [inline api/src/opentrons/hardware_control/gripper.py:36]: If we hold a reference to the config, why do we need to have references to members of the config?

**Review by amitlissack — COMMENTED** [review]
Just a couple questions.

File: api/src/opentrons/hardware_control/gripper.py (hunk 2 above)
Comment (ahiuchingau) [reply api/src/opentrons/hardware_control/gripper.py:83]: We don't need it yet, not until we need to save the config as a json file on the robot anyway. I pretty much copied all of these methods from the Pipette class

File: api/src/opentrons/hardware_control/gripper.py (hunk 3 above)
Comment (ahiuchingau) [reply api/src/opentrons/hardware_control/gripper.py:36]: True, i'll delete them

File: api/src/opentrons/hardware_control/instrument_abc.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+from opentrons.types import Point
+from .types import CriticalPoint
+
+
+class AbstractInstrument(ABC):
```
Comment (sfoster1) [inline api/src/opentrons/hardware_control/instrument_abc.py]: we probably want name, property, and model and so on, right?

**Review by sfoster1 — CHANGES_REQUESTED** [review]
This is an awesome step forward! And a really great way to get started by keeping it small.

Can we add some super basic tests - just checking that config and name information is preserved, and checking the current basic critical point implementation - so we begin as we mean to go on?

**Review by sfoster1 — APPROVED** [review]
Nice, love it.

**Pull request merged**